test/e2e/templates_test.go

```diff
@@ -0,0 +1,138 @@
+	require.NoError(t, err)
+
+	oauthConfigCopy := oauthConfig.DeepCopy()
+
+	oauthConfigCopy.Spec.Templates.Login.Name = "login"
+	oauthConfigCopy.Spec.Templates.ProviderSelection.Name = "providers"
+	oauthConfigCopy.Spec.Templates.Error.Name = "error"
+
+	oauthConfigCopy.Spec.IdentityProviders = []configv1.IdentityProvider{
+		{
+			Name: "super duper production-ready htpasswd idp",
+			IdentityProviderConfig: configv1.IdentityProviderConfig{
+				Type: configv1.IdentityProviderTypeHTPasswd,
+				HTPasswd: &configv1.HTPasswdIdentityProvider{
+					FileData: configv1.SecretNameReference{
+						Name: "htpasswd1",
+					},
+				},
+			},
+		},
+	}
+
+	_, err = configClient.ConfigV1().OAuths().Update(context.TODO(), oauthConfigCopy, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	defer func() {
+		oauthConfigNew, err := configClient.ConfigV1().OAuths().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		require.NoError(t, err)
+
+		oauthConfigNew.Spec.IdentityProviders = oauthConfig.Spec.IdentityProviders
+		oauthConfigNew.Spec.Templates = oauthConfig.Spec.Templates
+
+		_, err = configClient.ConfigV1().OAuths().Update(context.TODO(), oauthConfigNew, metav1.UpdateOptions{})
+		require.NoError(t, err)
+	}()
+
+	// wait for new rollout
+	err = e2e.WaitForClusterOperatorProgressing(t, configClient.ConfigV1(), "authentication")
+	require.NoError(t, err, "authentication operator never became progressing")
+
+	err = e2e.WaitForClusterOperatorAvailableNotProgressingNotDegraded(t, configClient.ConfigV1(), "authentication")
+	require.NoError(t, err, "failed to wait for the authentication operator to become available")
+
+	route, err := routeClient.RouteV1().Routes("openshift-authentication").Get(context.TODO(), "oauth-openshift", metav1.GetOptions{})
+	require.NoError(t, err)
+	oauthURL, err := url.Parse("https://" + route.Spec.Host)
+	require.NoError(t, err)
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true, // we don't care about certs in this test
+			},
+		},
+	}
+	oauthURL.Path = "/oauth/token/request" // should redirect to where the providers are
+	resp, err := httpClient.Get(oauthURL.String())
+	require.NoError(t, err)
+	body, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Contains(t, string(body), "provider selection test")
+
+	// modify URL to go to the IdP login
+	oauthURL.Path = "/oauth/token/display"
+	oauthURL.RawQuery = fmt.Sprintf("client_id=openshift-browser-client&idp=super+duper+production-ready+htpasswd+idp&redirect_uri=%s&response_type=code", url.QueryEscape(oauthURL.String()))
+	oauthURL.Path = "/oauth/authorize"
+	resp, err = httpClient.Get(oauthURL.String())
+	require.NoError(t, err)
+	body, err = ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Contains(t, string(body), "login test")
```

Comment:
Just wondering why the case to test `error.html` was not added?

Reply:
I was not sure how to force an error on the server, it did not feel so important with the other tests and the payload not erroring.